packages/backend/src/file-import/file-import.service.spec.ts

```
@@ -0,0 +1,384 @@
+  // one row too many and `importTable` drops it, reporting a file that fitted
+  // as truncated and losing a record.
+  it('sends no more rows than the cell cap can hold', async () => {
+    const columns = 25;
+    const budget = 50_000 / columns;
+    const header = Array.from({ length: columns }, (_, c) => `c${c}`).join(',');
+    // One row past the budget, so the limit is what stops it, not the file.
+    const rows = Array.from({ length: budget }, (_, r) =>
+      Array.from({ length: columns }, (_, c) => `v${r}_${c}`).join(','),
+    ).join('\n');
+
+    const result = await makeService(`${header}\n${rows}\n`).preview(WS, FILE_ID);
```

ESLint

[error] Replace `WS,·FILE_ID` with `⏎······WS,⏎······FILE_ID,⏎····`

(prettier/prettier)